### PR TITLE
MULE-9615: Extension configs can not define the same Source/Operation…

### PR DESCRIPTION
**Pull request by marianogonzalez**
… Class

### CODE REVIEW

File: core-tests/src/test/java/org/mule/runtime/core/util/ReferenceTestCase.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    public void nullValueHashCode()
+    {
+        assertThat(new Reference<>(null).hashCode(), is(0));
+    }
+
+    @Test
+    public void equalReferences()
+    {
+        final Object value = new Object();
+        assertEquals(new Reference<>(value), new Reference<>(value));
```
Comment (elrodro83) [inline core-tests/src/test/java/org/mule/runtime/core/util/ReferenceTestCase.java]: hamcrest

File: core-tests/src/test/java/org/mule/runtime/core/util/ReferenceTestCase.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    }
+
+    @Test
+    public void equalNullReferences()
+    {
+        assertEquals(new Reference<>(null), new Reference<>(null));
```
Comment (elrodro83) [inline core-tests/src/test/java/org/mule/runtime/core/util/ReferenceTestCase.java]: hamcrest

File: core-tests/src/test/java/org/mule/runtime/core/util/ReferenceTestCase.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    }
+
+    @Test
+    public void equalMixedReferences()
+    {
+        assertNotEquals(new Reference<>(null), new Reference<>(new Object()));
```
Comment (elrodro83) [inline core-tests/src/test/java/org/mule/runtime/core/util/ReferenceTestCase.java]: hamcrest

File: core-tests/src/test/java/org/mule/runtime/core/util/ReferenceTestCase.java [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        assertNotEquals(new Reference<>(new Object()), new Reference<>(null));
+    }
+
+    @Test
+    public void notEqualReferences()
+    {
+        assertNotEquals(new Reference<>(new Object()), new Reference<>(new Object()));
```
Comment (elrodro83) [inline core-tests/src/test/java/org/mule/runtime/core/util/ReferenceTestCase.java]: hamcrest

File: core-tests/src/test/java/org/mule/runtime/core/util/ReferenceTestCase.java (hunk 2 above)
Comment (marianogonzalez) [inline core-tests/src/test/java/org/mule/runtime/core/util/ReferenceTestCase.java]: this is a private method in the test which uses hamcrest. not the JUNIt one.

**Pull request merged**